src/plugins/embeddable/public/lib/embeddables/diff_embeddable_input.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import fastIsEqual from 'fast-deep-equal';
+import { pick, omit } from 'lodash';
+import { EmbeddableInput } from '.';
+
+// list out the keys from the EmbeddableInput type to allow lodash to pick them later
+const allGenericInputKeys: Readonly<Array<keyof EmbeddableInput>> = [
+  'lastReloadRequestTime',
+  'executionContext',
+  'searchSessionId',
+  'hidePanelTitles',
+  'disabledActions',
+  'disableTriggers',
+  'enhancements',
+  'syncColors',
+  'viewMode',
+  'title',
+  'id',
+] as const;
+
+const genericInputKeysToCompare = [
```

Comment:
It is possible to also use `Readonly<Array<keyof SOMETHING>>` here?

Reply:
It's actually implicitly typed due to the section underneath so that typescript throws errors if you try to put an element in the array which doesn't exist in EmbeddableInput. I did it this way to avoid listing out the keys twice.

``` typescript 
type GenericEmbedableInputToCompare = Pick<
  EmbeddableInput,
  typeof genericInputKeysToCompare[number]
>;
```

Do you think I should create a type with those keys explicitly picked, then create an array of keyof that type instead? In my opinion, they both work the same way, but this way gives less chance of error because it's defined only once.

Reply:
That's solves my concern 👍 
looks good